game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/game/selector/GameSelectorModel.java

```diff
@@ -226,7 +226,7 @@ public void loadDefaultGameSameThread() {
   private static Path getDefaultGameRealPath(final URI defaultGame) {
     try {
```

Comment:
@trevan  , could you remind me why we have the try/catch blocks in this method? It looks like they can be removed entirely. Is there an undocumented side-effect that we are relying on for this method to work properly?

Reply:
If the path is a zip file, then the `getFileSystem` will initially through `FileSystemNotFoundException` so you'll have to call `newFileSystem` to create it.  The second time, `getFileSystem` will not have any problem.  For non zip, `getFileSystem` doesn't through anything.  This is documented at https://docs.oracle.com/javase/7/docs/api/java/nio/file/FileSystems.html#getFileSystem(java.net.URI).

I don't know why it would throw null pointer.  Maybe that happens if `defaultGame` is a null uri?

Reply:
Could we call 'newFileSystem' first always and then call 'getFileSystem'?

It looks like if the URI is missing a scheme, then we'll get a NPE in file systems.

Reply:
I force pushed over this PR and drastically changed the solution, just catching a NPE is a bad fix.

I think we probably could use some documentation to explain the needed side effects from the FileSystem calls, I was certainly confused by it and wondered why it was needed. 

FWIW, I think we've made good progress here, seems we have much more robust code here that doesn't break.